Build failure with macOS 10.13.5 and CHPL_DEVELOPER=true

### Summary of Problem
When running `make` on a clean repo on macOS 10.13.5 with Xcode 9.4.1 and CHPL_DEVELOPER=true, the compiler build fails in codegen.

Unsetting CHPL_DEVELOPER allows build to succeed, because the error is reduced to a warning.

### Steps to Reproduce

* checkout master branch and update to latest (commit: e98051e449897d9fcca5a71c31d48fa82e109c85)
* `make clobber && git clean -dxf`
* `make`
* Observe the following error in codegen/

```
***** codegen/ *****
clang++ -c -std=gnu++11  -MMD -MP -g -Wall -Werror -Wpointer-arith -Wwrite-strings -Wno-strict-aliasing -Wmissing-declarations -I. -I../include/darwin -I../include    -o ../../build/compiler/darwin/clang/llvm-none/codegen/symbol.o symbol.cpp
clang++ -c -std=gnu++11  -MMD -MP -g -Wall -Werror -Wpointer-arith -Wwrite-strings -Wno-strict-aliasing -Wmissing-declarations -I. -I../include/darwin -I../include    -o ../../build/compiler/darwin/clang/llvm-none/codegen/type.o type.cpp
clang++ -c -std=gnu++11  -MMD -MP -g -Wall -Werror -Wpointer-arith -Wwrite-strings -Wno-strict-aliasing -Wmissing-declarations -I. -I../include/darwin -I../include    -o ../../build/compiler/darwin/clang/llvm-none/codegen/codegen.o codegen.cpp
In file included from codegen.cpp:20:
In file included from ../include/codegen.h:24:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:442:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__tree:1819:22: error: the specified
      comparator type does not provide a const call operator [-Werror,-Wuser-defined-warnings]
                     __trigger_diagnostics()), "");
                     ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/set:400:28: note: in instantiation of member
      function 'std::__1::__tree<TypeSymbol *, compareSymbolFunctor, std::__1::allocator<TypeSymbol *> >::~__tree' requested here
class _LIBCPP_TEMPLATE_VIS set
                           ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__tree:970:7: note: from 'diagnose_if'
      attribute on '__trigger_diagnostics':
      _LIBCPP_DIAGNOSE_WARNING(!__invokable<_Compare const&, _Tp const&, _Tp const&>::value,
      ^                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__config:1095:20: note: expanded from macro
      '_LIBCPP_DIAGNOSE_WARNING'
    __attribute__((diagnose_if(__VA_ARGS__, "warning")))
                   ^           ~~~~~~~~~~~
1 error generated.
make[3]: *** [../../build/compiler/darwin/clang/llvm-none/codegen/codegen.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[2]: *** [codegen.makedir] Error 2
make[1]: *** [compiler] Error 2
make: *** [comprt] Error 2
```

### Configuration Information

- Output of `chpl --version`: n/a
- Output of `$CHPL_HOME/util/printchplenv --anonymize`:

```
CHPL_TARGET_PLATFORM: darwin
CHPL_TARGET_COMPILER: clang
CHPL_TARGET_ARCH: native
CHPL_LOCALE_MODEL: flat
CHPL_COMM: none
CHPL_TASKS: qthreads
CHPL_LAUNCHER: none
CHPL_TIMERS: generic
CHPL_UNWIND: none
CHPL_MEM: jemalloc
CHPL_ATOMICS: intrinsics
CHPL_GMP: gmp
CHPL_HWLOC: hwloc
CHPL_REGEXP: re2
CHPL_AUX_FILESYS: none
```
- Back-end compiler and version, e.g. `gcc --version` or `clang --version`:

```
$ clang --version
Apple LLVM version 9.1.0 (clang-902.0.39.2)
Target: x86_64-apple-darwin17.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

$ clang++ --version
Apple LLVM version 9.1.0 (clang-902.0.39.2)
Target: x86_64-apple-darwin17.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

- (For Cray systems only) Output of `module list`: n/a

--- a/compiler/codegen/codegen.cpp
+++ b/compiler/codegen/codegen.cpp
@@ -57,7 +57,7 @@
 #include <vector>
 
 // function prototypes
-static bool compareSymbol(void* v1, void* v2);
+static bool compareSymbol(const void* v1, const void* v2);
 
 // Global so that we don't have to pass around
 // to all of the codegen() routines
@@ -272,7 +272,7 @@ genClassIDs(std::vector<TypeSymbol*> & typeSymbol, bool isHeader) {
 
 struct compareSymbolFunctor {
   // This is really operator less-than
-  bool operator() (Symbol* a, Symbol* b) {
+  bool operator() (const Symbol* a, const Symbol* b) const {
     return compareSymbol(a, b);
   }
 };
@@ -892,7 +892,7 @@ genClassNames(std::vector<TypeSymbol*> & typeSymbol, bool isHeader) {
 
 
 static bool
-compareSymbol(void* v1, void* v2) {
+compareSymbol(const void* v1, const void* v2) {
   Symbol* s1 = (Symbol*)v1;
   Symbol* s2 = (Symbol*)v2;
   ModuleSymbol* m1 = s1->getModule();

